packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -320,31 +335,94 @@ function WorklistDataGrid(props) {
+      message = intl.formatMessage({ id: 'Terra.worklist-data-grid.all-rows-unselected' });
       // Esc (while in row selection mode and rows are selected): Clear selection
       if (onClearSelectedRows) {
         onClearSelectedRows();
       }
     } else if (onDisableSelectableRows) {
-      setAriaLiveMessage(intl.formatMessage({ id: 'Terra.worklist-data-grid.row-selection-mode-disabled' }));
+      rowSelection.current.rangeUsingFocusedRow = null;
+      message = intl.formatMessage({ id: 'Terra.worklist-data-grid.row-selection-mode-disabled' });
       onDisableSelectableRows();
     }
+    setAriaLiveMessage(message);
+  };
+
+  const selectMultipleRows = (newEndRowIndex, rangeInfo) => {
+    if (!onRowSelect || !rangeInfo) {
+      return;
+    }
+
+    if (!hasSelectableRows) {
+      onEnableRowSelection();
+    }
+
+    // Rows in range to remain selected
+    const selectionStartRowIndex = Math.min(rangeInfo.anchorRow, newEndRowIndex);
+    const selectionEndRowIndex = Math.max(rangeInfo.anchorRow, newEndRowIndex);
+
+    // We are subtracting 1 to accommodate for the column header in the grid.
+    const rowIdsToSelect = rows.slice(selectionStartRowIndex - 1, selectionEndRowIndex).map(row => row.id);
+
+    // Determine if there are rows that are no longer in range that need to be unselected.
+    let rowIdsToUnselect = [];
+    if (rangeInfo.anchorRow > rangeInfo.previousSelectionEndRow) {
+      // The range extends upward from the anchor row
+      if (newEndRowIndex > rangeInfo.previousSelectionEndRow) {
+        // The range was moved down towards the anchor so rows that no longer qualify for the range need to be unselected.
+        rowIdsToUnselect = rows.slice(rangeInfo.previousSelectionEndRow - 1, Math.min(newEndRowIndex, rangeInfo.anchorRow)).map(row => row.id);
+      }
+    } else if (rangeInfo.anchorRow < rangeInfo.previousSelectionEndRow) {
+      // The range extends downward from the anchor row
+      if (newEndRowIndex < rangeInfo.previousSelectionEndRow) {
+        // The range was moved up towards the anchor so rows that no longer qualify for the range need to be unselected. New endRangeIndex becomes ordered End
+        rowIdsToUnselect = rows.slice(Math.max(rangeInfo.anchorRow, newEndRowIndex), rangeInfo.previousSelectionEndRow).map(row => row.id);
+      }
+    }
+
+    const ariaMessage = `${intl.formatMessage({
+      id: hasSelectableRows
+        ? 'Terra.worklist-data-grid.row-selection-multiple-rows-selected'
+        : 'Terra.worklist-data-grid.row-selection-mode-enabled',
+    })}. ${intl.formatMessage({
+      id: rowIdsToSelect.length === 1
+        ? 'Terra.worklist-data-grid.row-selection-template'
+        : 'Terra.worklist-data-grid.row-selection-selected-rows-range',
+    }, { row: rangeInfo.anchorRow, endRow: newEndRowIndex })}`;
+
+    setAriaLiveMessage(ariaMessage);
+    onRowSelect(rowIdsToSelect, rowIdsToUnselect);
```

Comment:
I would expect that we would only be passing one array back to consumers.  We should not need to pass two separate arrays.

Reply:
We need not, but we chose to pass two separate arrays to make it clearer for the consumers.

Reply:
I still don't think having the two array is going to work out well, especially since we don't provide a selection state for a single row selection.

Reply:
Concurring with @cm9361 Because we could just manage the selected state with 1 array. Only care about that one, the others we could just assume are now unchecked. I think that's a simpler way to approach this.